### PR TITLE
flent: convert to unittestCheckHook and add passthru.updateScript

### PR DESCRIPTION
**Pull request by mmlb**
Convert to plain old unittestCheckHook to simplify test setup.
Add passthru.updateScript so @r-ryantm bot can update the versions.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mmlb** [comment]
Not sure why eval is failing complaining about fetchPypi missing, its there and works on my machine :tm: :crying_cat_face:

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build flent

File: pkgs/by-name/fl/flent/package.nix [hunk 1]
```diff
@@ -56,11 +43,13 @@ python3Packages.buildPythonApplication rec {
...
   '';
 
+  passthru.updateScript = nix-update-script { };
   meta = {
     description = "FLExible Network Tester";
     homepage = "https://flent.org";
     license = lib.licenses.gpl3;
     maintainers = with lib.maintainers; [ mmlb ];
     mainProgram = "flent";
+    platforms = lib.platforms.linux;
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/flent/package.nix:53]: Flent supports macOS according to https://pypi.org/project/flent . Why is this?

Comment (mmlb) [reply pkgs/by-name/fl/flent/package.nix:53]: This package has never been enabled for macos due to xvfb-run, now that I removed that the tests fail so I explicitly limited to linux for now and will try to figure the tests issue on macos layer.

Comment (Aleksanaa) [reply pkgs/by-name/fl/flent/package.nix:53]: Would be better to mark `badPlatforms = lib.platforms.darwin` then.

Comment (mmlb) [reply pkgs/by-name/fl/flent/package.nix]: Thanks for the suggestion, updated.

File: pkgs/by-name/fl/flent/package.nix [hunk 2]
```diff
@@ -56,11 +43,14 @@ python3Packages.buildPythonApplication rec {
...
   '';
 
+  passthru.updateScript = nix-update-script { };
   meta = {
     description = "FLExible Network Tester";
     homepage = "https://flent.org";
     license = lib.licenses.gpl3;
     maintainers = with lib.maintainers; [ mmlb ];
     mainProgram = "flent";
+    platforms = lib.platforms.all;
```
Comment (getchoo) [inline pkgs/by-name/fl/flent/package.nix:53]:
```suggestion
```
`buildPythonApplication` already [sets](https://github.com/NixOS/nixpkgs/blob/61194a6d4b7dc5fb8faf8aaa4f802a7063bbe774/pkgs/development/interpreters/python/mk-python-derivation.nix#L328) a proper default

**Review by getchoo — DISMISSED** [review]

File: pkgs/by-name/fl/flent/package.nix (hunk 2 above)
Comment (mmlb) [reply pkgs/by-name/fl/flent/package.nix:53]: Nice!

Comment (mmlb) [reply pkgs/by-name/fl/flent/package.nix]: Updated

**Comment by Aleksanaa** [comment]
@ofborg build flent

**Comment by mmlb** [comment]
@Aleksanaa are you able to merge?

**Pull request merged**